src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardIdentification.cs

```diff
@@ -0,0 +1,337 @@
+        public string ProcessorName
+        {
+            get
+            {
+                _settings.TryGetValue("Hardware", out string hardware);
+                return hardware;
+            }
+        }
+
+        /// <summary>
+        /// Gets the board firmware version.
+        /// </summary>
+        public int Firmware
+        {
+            get
+            {
+                if (_settings.TryGetValue("Revision", out string revision)
+                    && !string.IsNullOrEmpty(revision)
+                    && int.TryParse(revision, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out int firmware))
+                {
+                    return firmware;
+                }
+
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the serial number.
+        /// </summary>
+        public string SerialNumber
+        {
+            get
+            {
+                if (_settings.TryGetValue("Serial", out string serial)
+                    && !string.IsNullOrEmpty(serial))
+                {
+                    return serial;
+                }
+
+                return string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether board is overclocked.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if board is overclocked; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsOverclocked
+        {
+            get
+            {
+                var firmware = Firmware;
+                return (firmware & 0xFFFF0000) != 0;
+            }
+        }
+
+        #endregion
+
+        #region Private Helpers
+
+        private Processor LoadProcessor(Model model)
+        {
+            switch (model)
+            {
+                case Model.RaspberryPiA:
+                case Model.RaspberryPiAPlus:
+                case Model.RaspberryPiBRev1:
+                case Model.RaspberryPiBRev2:
+                case Model.RaspberryPiBPlus:
+                case Model.RaspberryPiComputeModule:
+                case Model.RaspberryPiZero:
+                case Model.RaspberryPiZeroW:
+                    return Processor.Bcm2708;
+                case Model.RaspberryPi2B:
+                // TBC: B3(+) should be a BCM2710 processor ...
+                case Model.RaspberryPi3B:
+                case Model.RaspberryPi3BPlus:
+                case Model.RaspberryPiComputeModule3:
+                    return Processor.Bcm2709;
+                case Model.RaspberryPi4:
+                    return Processor.Bcm2711;
+                default:
+                    return Processor.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Detect the board CPU information from /proc/cpuinfo
+        /// </summary>
+        /// <returns>
+        /// The <see cref="RaspberryBoardIdentification"/>.
+        /// </returns>
+        internal static RaspberryBoardIdentification LoadBoard()
```

Comment:
If you rename the class the more appropiate name for this would be `LoadBoardInfo`